packages/frontend/src/features/sqlRunner/duckDBQuery.ts

```diff
@@ -26,15 +26,24 @@ function initializeBundle() {
 }
 initializeBundle();
 
-export const duckDBFE: DuckDBSqlFunction = async (sql, rowData) => {
+export const duckDBFE: DuckDBSqlFunction = async (sql, rowData, columns) => {
     const bundle = await bundlePromise;
-    const arrowTable = tableFromJSON(rowData);
     const worker = new Worker(bundle.mainWorker!);
     const logger = new duckdb.ConsoleLogger();
     const db = new duckdb.AsyncDuckDB(logger, worker);
     await db.instantiate(bundle.mainModule, bundle.pthreadWorker);
     const conn = await db.connect();
 
+    const columnReferences = columns.map((c) => c.reference);
+    const typedArrays = columnReferences.reduce((acc, c) => {
+        const col = rowData.map((r) => r[c]);
+        return {
+            ...acc,
+            [c]: col,
+        };
+    }, {});
```

Comment:
by creating a typed array https://arrow.apache.org/docs/js/functions/Arrow_dom.tableFromArrays.html `null` all-null-values-tables won't be removed. 

I've kept the existing logic to disable these columns from being used - I believe this is fine. 

Regarding your point on handling the error @ZeRego, I believe this one is fixed - I tested example 1

Reply:
Both example 1 and 2 are fixed. But I was able to reproduce the error with a different scenario:

1. Run query ( all columns have values )
2. Edit chart x-axis to X
3. Edit query to not return column X
4. See chart error

<img width="1483" alt="Screenshot 2024-08-06 at 10 08 49" src="https://github.com/user-attachments/assets/8f37c3f9-02f3-4895-9f67-e7050652c559">

Reply:
will do that! 
I also tried to find an official error parser but no luck :/

Reply:
<img width="1905" alt="Screenshot 2024-08-06 at 10 20 16" src="https://github.com/user-attachments/assets/68874520-0758-4ec6-96f2-3c2d5c95047e">

what do you think?